# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/nsPresContext.cpp
# Commit: 8ed7c109c239
# Full Hash: 8ed7c109c239b07a02e7f3139f075c8f1d3ff0a0
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-09-20 21:52:25
# Description:
#   Bug 1580690 - Ensure src:local() entries in the user font set are refreshed if the platform font list is rebuilt. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46580
# ==============================================================================

diff -r 932d3794c5d4 -r 8ed7c109c239 layout/base/nsPresContext.cpp
--- a/layout/base/nsPresContext.cpp	Thu Sep 19 17:49:39 2019 +0000
+++ b/layout/base/nsPresContext.cpp	Fri Sep 20 15:24:33 2019 +0000
@@ -138,6 +138,12 @@
   // stale nsFontMetrics objects from it.
   DeviceContext()->FlushFontCache();
 
+  // If there's a user font set, discard any src:local() faces it may have
+  // loaded because their font entries may no longer be valid.
+  if (Document()->GetFonts()) {
+    Document()->GetFonts()->GetUserFontSet()->ForgetLocalFaces();
+  }
+
   // We can trigger reflow by pretending a font.* preference has changed;
   // this is the same mechanism as gfxPlatform::ForceGlobalReflow() uses
   // if new fonts are installed during the session, for example.
